### PR TITLE
Implemented the location module

### PR DESCRIPTION
**Pull request by dineshGramenerit**
- Designed location list screen and integrated the list API.
- Designed location ADD and EDIT screens and integrated the ADD and EDIT APIs.
- Added the form validations.
- Implemented the export (PDF, EXCEL).
- Implemented the search functionality.
- Added the infinity scroll option.

### CODE REVIEW

File: resource_management_mobile/src/app/pages/location/location.page.html [hunk 1]
```diff
@@ -1,3 +1,94 @@
...
+            </ion-item-option>
+          </ion-item-options>
+        </ion-item-sliding>
+      </ion-list>
+      <ion-infinite-scroll>
+        <ion-infinite-scroll-content loadingText="Please wait..."
+          loadingSpinner="bubbles"></ion-infinite-scroll-content>
+      </ion-infinite-scroll>
+      <ion-list class="nodata-found" lines="none" *ngIf="locationData.length==0">
+        <h5>Not Data Found!</h5>
```
Comment (deviprasad0327) [inline resource_management_mobile/src/app/pages/location/location.page.html]: Change this to "No Data Found!"

File: resource_management_mobile/src/app/pages/location/location.page.ts [hunk 2]
```diff
@@ -1,15 +1,244 @@
...
+    this.locationService.deleteLocation(id).subscribe({
+      next: (response) => {
+        response = response as locationResponce;
+        this.locationData.splice(index, 1);
+        this.toastService.presentToast(
+          this.toastConstants.Delete_success_message
+        );
+      },
+      error: (response) => {
+        this.toastService.presentToast(this.toastConstants.try_again);
```
Comment (deviprasad0327) [inline resource_management_mobile/src/app/pages/location/location.page.ts]: Instead of "presentToast" use "errorToast" from toast service.

**Review by deviprasad0327 — COMMENTED** [review]
Kindly check mention comments

File: resource_management_mobile/src/app/pages/location/location.page.ts (hunk 2 above)
Comment (dineshGramenerit) [reply resource_management_mobile/src/app/pages/location/location.page.ts]: changed

File: resource_management_mobile/src/app/pages/location/location.page.html (hunk 1 above)
Comment (dineshGramenerit) [reply resource_management_mobile/src/app/pages/location/location.page.html]: changed

**Review by deviprasad0327 — APPROVED** [review]

**Pull request merged**